### PR TITLE
Remove using PHP internal function print_r 

### PR DESCRIPTION
**Pull request by xmorave2**
When print_r is called on some non-trivial object (like db row objects), it could easily exhaust all memory and the process then ends with error just trying to log error (or even debug!). 

I used symfony/var-dumper to avoid this.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -96,7 +96,8 @@
         "webfontkit/open-sans": "^1.0",
         "webmozart/glob": "^4.7",
         "wikimedia/composer-merge-plugin": "2.1.0",
-        "yajra/laravel-pdo-via-oci8": "3.4.2"
+        "yajra/laravel-pdo-via-oci8": "3.4.2",
+        "symfony/var-dumper": "6.4.6"
```
Comment (demiankatz) [inline composer.json]: We should move this line up to keep the dependency list in alphabetical order.

File: module/VuFind/tests/unit-tests/src/VuFindTest/Log/VarDumperTraitTest.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+            'string' => [
+                'some_string',
+                '"some_string"\n',
+            ],
+            'array' => [
+                [42, 'string', false],
+                'array:3 \[\n  0 => 42\n  1 => "string"\n  2 => false\n\]\n',
+            ],
+            'object' => [
+                new TestClass(42, 'string', false),
```
Comment (demiankatz) [inline module/VuFind/tests/unit-tests/src/VuFindTest/Log/VarDumperTraitTest.php]: Did you consider using an anonymous class here to eliminate the need for a separate TestClass.php file?

File: module/VuFind/src/VuFind/View/Helper/Root/RecordLinker.php [hunk 3]
```diff
@@ -48,6 +48,8 @@
  */
 class RecordLinker extends \Laminas\View\Helper\AbstractHelper
 {
+    use \VuFind\Log\VarDumperTrait;
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/View/Helper/Root/RecordLinker.php]: Is there a reason you're adding the trait here but not changing any other code?

File: module/VuFindSearch/src/VuFindSearch/Backend/SRU/Connector.php [hunk 4]
```diff
@@ -221,7 +221,7 @@ protected function call($method = 'GET', $params = null, $process = true)
             $queryString = implode('&', $query);
         }
 
-        $this->debug('Connect: ' . print_r($this->host . '?' . $queryString, true));
+        $this->debug('Connect: ' . $this->varDump($this->host . '?' . $queryString));
```
Comment (demiankatz) [inline module/VuFindSearch/src/VuFindSearch/Backend/SRU/Connector.php]: This looks like a plain string; should we omit the varDump entirely?

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks, @xmorave2 -- this looks pretty good to me, but I had a few small questions and comments. See below:

File: composer.json (hunk 1 above)
Comment (xmorave2) [reply composer.json]: Done

File: module/VuFind/tests/unit-tests/src/VuFindTest/Log/VarDumperTraitTest.php (hunk 2 above)
Comment (xmorave2) [reply module/VuFind/tests/unit-tests/src/VuFindTest/Log/VarDumperTraitTest.php]: Good point, I changed it.

File: module/VuFind/src/VuFind/View/Helper/Root/RecordLinker.php (hunk 3 above)
Comment (xmorave2) [reply module/VuFind/src/VuFind/View/Helper/Root/RecordLinker.php]: It's a mistake, fixed.

File: module/VuFindSearch/src/VuFindSearch/Backend/SRU/Connector.php (hunk 4 above)
Comment (xmorave2) [reply module/VuFindSearch/src/VuFindSearch/Backend/SRU/Connector.php]: Yes, there is no need for dumping a string, fixed

**Comment by xmorave2** [comment]
@demiankatz Thanks for the comments, I updated the code to reflect them.

**Review by demiankatz — APPROVED** [review]
Thanks, @xmorave2 -- this all looks good to me, and tests are passing now. The only issue I ran into is that Composer was complaining about the lock file being out of sync with composer.json. I ran `composer update` to get things back in sync and pushed up the results.

**Pull request merged**